thirdparty-api/data-models.md

```diff
@@ -0,0 +1,953 @@
+
+###### 3.1.1.2.2  `PUT /accounts/<ID>/error`
+
+Used by: DFSP
+
+The `PUT /accounts/<ID>/error` response is used to inform the requester that an account list
+request has given rise to an error. The identifier ID given in the call are
+the values given in the original request (see Section 3.1.1.1.1 above.)
+
+The data content of the message is given below.
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| errorInformation | 1 | ErrorInformation | Information describing the error and error code. |
+
+#### 3.1.2 `/consentRequests`
+
+The `/consentRequests` resource is used by a PISP to initiate the process of linking with a DFSP’s 
+account on behalf of a user. The PISP contacts the DFSP and sends a list of the permissions that 
+it wants to obtain and the accounts for which it wants permission.
+
+##### 3.1.2.1 Requests
+
+This section describes the services that can be requested by a client on the API resource 
+`/consentRequests`.
+###### 3.1.2.1.1 `GET /consentRequests/<ID>`
+
+Used by: PISP
+
+The HTTP request `GET /consentRequests/<ID>` is used to get information about a previously 
+requested consent. The `<ID>` in the URI should contain the requestId that was assigned to the 
+request by the PISP when the PISP originated the request.
+
+Callback and data model information for `GET /consentRequests/<ID>`:
+- Callback – `PUT /consentRequests/<ID>`
+- Error Callback – `PUT /consentRequests/<ID>/error`
+- Data Model – Empty body
+
+###### 3.1.2.1.2 `POST /consentRequests`
+
+Used by: PISP
+
+The HTTP request `POST /consentRequests` is used to request a DFSP to grant access to one or more
+accounts owned by a customer of the DFSP for the PISP who sends the request.
+
+Callback and data model for `POST /consentRequests`:
+- Callback: `PUT /consentRequests/<ID>`
+- Error callback: `PUT /consentRequests/<ID>/error`
+- Data model – see below
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| consentRequestId | 1 | CorrelationId | Common ID between the PISP and the Payer DFSP for the consent request object. The ID should be reused for resends of the same consent request. A new ID should be generated for each new consent request. |
+| userId           | 1 | string        | The identifier used in the `GET /accounts/<ID>`. Used by the DFSP to correlate an account lookup to a `consentRequest` |
+| scopes | 1..n | Scope | One or more requests for access to a particular account. In each case, the address of the account and the types of access required are given. |
+| authChannels | 1..n | ConsentRequestChannelType | A collection of the types of authentication that the DFSP may use to verify that its customer has in fact requested access for the PISP to the accounts requested. |
+| callbackUri | 1 | Uri |The callback URI that the user will be redirected to after completing verification via the WEB authorization channel |
```

Comment:
Yes. The PISP must provide a callbackUri because it doesn't know ahead of time which AuthChannel the DFSP will choose to use. I'll add a note to that effect